saleor/graphql/product/mutations/products.py

```diff
@@ -334,6 +360,20 @@ class Meta:
         description = 'Updates an existing product.'
         model = models.Product
 
+    @classmethod
+    @transaction.atomic
+    def save(cls, info, instance, cleaned_input):
+        instance.save()
+        if not instance.product_type.has_variants:
+            variant = instance.variants.first()
+            if 'track_inventory' in cleaned_input:
+                variant.track_inventory = cleaned_input.get('track_inventory')
+            if 'quantity' in cleaned_input:
+                variant.quantity = cleaned_input.get('quantity')
+            if 'sku' in cleaned_input:
```

Comment:
I think that this whole logic should live in `clean_input` instead, ane here we should just assign the values to `variant` instance.